chef/automate
# Notifications service compliance update

### :nut_and_bolt: Description: What code changed, and why?
InSpec reports have been updated to have truncated results. This PR is updating the number of control tests based on truncated results added structure from this PR https://github.com/chef/automate/pull/3595. 

A bug was found and fixed with this work. The number of critical failed control tests was not counted correctly. The code was instead counting the number of critical failed controls. 

### :chains: Related Resources
https://github.com/chef/automate/issues/3782

### :athletic_shoe: How to Build and Test the Change
1. `build components/notifications-service && build components/compliance-service && start_all_services`
1. Create two InSpec notifications. One for slack and webhook. One way to create a webhook endpoint is with the http://requestbin.sjcmmsn.com/ site. I used my personal slack webhook to test it. 
1. Send a failed InSpec report with the removed test structure `JSON_FILE=/src/components/notifications-client/builder/testdata/inspec-removed-controls-failure.json send_inspec_example`
1. In the webhook message ensure the "number_of_tests" in the control is 311 and the "number_of_failed_tests" is 154. Here is a link to the entire message https://gist.github.com/lancewf/60fc39cb2f31ccfe87832a31ab96ae8a
1. Ensure the slack message looks like the image below. 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
![image](https://user-images.githubusercontent.com/1679247/83931300-cacee380-a750-11ea-89e8-a6c5742a5218.png)
